src/main/java/ObjectPool.java

```diff
@@ -0,0 +1,81 @@
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+
+public abstract class ObjectPool<T> implements Pool<T> {
```

Comment:
You specify a very anonymous T type. In theory this can become any object. However later in the code you implicitly limit this to a ViRedis. If somebody sets this class up with String as type you may get a lot of runtime (or compiletime) exceptions.
Consider removing the type all together (and specifying ViRedis as type to pass to Pool) or specify <T extends ViRedis> if ViRedis is extensible. If you do the latter you will have to reconsider the createNew method as well (as that creates a direct ViRedis).

Reply:
done!